site/content/blog/2023-09-11-browser-recorder/index.md

```diff
@@ -0,0 +1,127 @@
+
+## Integration with Zest add-on 
+
+The [Zest add-on](/docs/desktop/addons/zest/) is extended to provide a new record type called `Client (browser) side script`. This record type is used to record the user interactions and generate the Zest script. The Zest add-on leverages the power of the Client add-on and the browser extension to record the user interactions and generate the Zest script. 
+
+## Features and Benefits
+
+The ZAP Browser Extension project offers a number of features and benefits, including:
+
+* User Interaction Recording: The browser extension will capture user interactions as they navigate through a website, effectively creating a trail of actions.
+* Automated Zest Script Generation: The captured interactions will be transformed into a Zest script in JSON format, ready for use with ZAP's testing capabilities.
+* Local Script Storage: Users will have the option to save generated scripts locally, fostering reusability and enhancing collaboration.
+* Seamless Integration with ZAP: The final deliverable of this project will be a fully functional browser extension that seamlessly integrates with ZAP, ensuring a smooth transition from interaction recording to vulnerability testing.
+
+## How can you use? 
+
+The ZAP Browser Recorder's first verison has been released and can be used by the users. We have released the browser extension for the following browsers:
+
+* Chrome (Under Review by the Chrome Web Store)
+* Firefox (Published)
+
+The browser recorder feature can be used in two ways: 
+
+### 1. Using the Zest add-on
+
+- First of all you need to install the Client and Zest add-on from the ZAP marketplace. 
+- Then you need to make sure that the browser extension has been enabled for Chrome as we currently only support integration with Chrome browser. 
+- Then you will have to go to the `Record a Zest Script` option as shown in the image below:
+
+![Zest Record Button](images/zest-record.png)
+
+- Then you will have to select the `Client (browser) side script` option and fill the initial URL which you wish to record as shown in the image below:
+
+![Record Zest Script Dialog](images/zest-record-dialog.png)
+
+- Now, click on Start Recording. The browser will open and you will be redirected to the URL you have entered. Now click on the extensions and you will see the ZAP browser extension. 
+- Click on the extension and you will see a popup window. Now click on the `Record` button to start the recording.
+- Now you can do whatever you want to record and when you are done with the recording you can click on the `Stop` button.
+- The script will be automatically generated by the Zest add-on and you can see the script in the Scripts -> Session Management Scripts section. 
+
+### 2. Using the Browser Extension
+
+- For ease, users can use the existing ZAP browsers from the [Quick Start](/docs/desktop/addons/quick-start/) section which will already have extension install.
```

Comment:
`… can launch browsers from ZAP using the [Quick Start add-on](…/#manual-explore), which will already have the extension installed.`